prog/minio/minio_cluster_nexus.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Minio::MinioClusterNexus < Prog::Base
+  subject_is :minio_cluster
+
+  semaphore :restart, :destroy
+
+  def self.assemble(project_id, cluster_name, location, admin_user,
+    storage_size_gib, pool_count, server_count, driver_count, vm_size)
+    unless (project = Project[project_id])
+      fail "No existing project"
+    end
+
+    Validation.validate_vm_size(vm_size)
+    Validation.validate_location(location, project.provider)
+    Validation.validate_name(cluster_name)
+    Validation.validate_minio_username(admin_user)
+
+    DB.transaction do
+      ubid = MinioCluster.generate_ubid
+
+      minio_cluster = MinioCluster.create(
+        name: cluster_name,
+        location: location,
+        admin_user: admin_user,
+        admin_password: SecureRandom.base64(15).gsub(/[+\/]/, "+" => "_", "/" => "-"),
```

Comment:
I'd use `SecureRandom.alphanumeric`. If you want even entropy, how about `SecureRandom.urlsafe_base64`?